src/main/java/com/commercetools/sync/sdk2/products/helpers/ProductBatchValidator.java

```diff
@@ -1,22 +1,27 @@
 package com.commercetools.sync.sdk2.products.helpers;
 
-import com.commercetools.sync.commons.helpers.BaseBatchValidator;
+import static com.commercetools.sync.commons.utils.ResourceIdentifierUtils.REFERENCE_ID_FIELD;
+import static com.commercetools.sync.commons.utils.ResourceIdentifierUtils.REFERENCE_TYPE_ID_FIELD;
+import static com.commercetools.sync.commons.utils.ResourceIdentifierUtils.isReferenceOfType;
+import static java.lang.String.format;
+import static java.util.Collections.emptySet;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import com.commercetools.api.models.product.Attribute;
+import com.commercetools.api.models.product.ProductDraft;
+import com.commercetools.api.models.product.ProductVariantDraft;
 import com.commercetools.sync.commons.utils.SyncUtils;
 import com.commercetools.sync.customobjects.helpers.CustomObjectCompositeIdentifier;
-import com.commercetools.sync.products.ProductSyncOptions;
 import com.commercetools.sync.products.helpers.VariantReferenceResolver;
+import com.commercetools.sync.sdk2.commons.helpers.BaseBatchValidator;
+import com.commercetools.sync.sdk2.products.ProductSyncOptions;
 import com.fasterxml.jackson.databind.JsonNode;
 import io.sphere.sdk.categories.Category;
 import io.sphere.sdk.customobjects.CustomObject;
 import io.sphere.sdk.products.Product;
```

Comment:
Can we migrate `Product` resource to sdk-v2 here? If it has dependencies with other resources migration at this moment. Please mark `TODO` here to avoid missing it.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/878/commits/2ea4a697e66475818aa6f95e3f8283fb02bbd3b4